test/mapwindow.jl

```diff
@@ -71,6 +72,19 @@ using ImageFiltering: IdentityUnitRange
         Amin = groundtruth(min, A, w)
         @test minval == Amin
     end
+    # offsets
+    @testset "offsets" begin
+        arrays = [rand(5), rand(5,5), rand(5,5,5)]
+        @testset "offsets ($f, $offset, $window, $dim)" for f in (extrema,maximum,minimum),
+                                                            offset in -5:5,
+                                                            window in [1:2:9; [0:2,-2:0]],
+                                                            (dim,a) in enumerate(arrays)
+            offsets = ntuple(_->offset,dim)
+            windows = ntuple(_->window,dim)
+            @test OffsetArray(mapwindow(f,a,windows),offsets) == mapwindow(f,OffsetArray(a,offsets),windows)
```

Comment:
I think this is a good test and we should have it. Just asking, is this test different from behavior on master? Could also add a case where there is no handcrafted version for `f`. E.g. `f = x -> sum(x)`

Reply:
The behavior on master is an assertion error when the input array has non-zero offset. 
The original version of this PR simply removed these assertions and loosened some type checks, which happens to work "accidentally", but would stop working once JuliaArrays/OffsetArrays.jl#104 is fixed (e.g. JuliaArrays/OffsetArrays.jl#105 is merged).
The current version is the opposite: it depends on JuliaArrays/OffsetArrays.jl#105 to work.

Not sure what you mean about handcrafted version for `f`. AFAICT there is no specialized implementation of `mapwindow` for `minimum` or `maximum`.

Reply:
Thanks for clarifying. I do not remember for which functions there is a specialized implementation. Even if there is none for `minimum/maximum` right now, it could potentially be added in future, so I believe adding a lambda f won't hurt.